discovery_engine_core/core/src/stack.rs

```diff
@@ -63,7 +63,7 @@ pub type BoxedOps = Box<dyn Ops + Send + Sync>;
 /// `Id` is used to connect [`Ops`](ops::Ops) with the corresponding data of a stack.
 #[derive(Serialize, Deserialize, Clone, Copy, Debug, PartialEq, Eq, Hash, From, Display)]
 #[cfg_attr(test, derive(Default))]
-pub struct Id(Uuid);
+pub struct Id(pub Uuid);
```

Comment:
we could put a `#[repr(transparent)]` on the `Id`, then we wouldn't have to make the wrapped field public. similarly for `document::Id`.

Reply:
I'm not sure if a `#[repr(transparent)]` which you are supposed to use to write the inner type without making the inner type public is an anit-pattern or not tbh. :thinking: 

It kinda makes the field pseudo public, which is why I think it might be an anit-pattern.

But then only for unsafe code, mainly ffi code, which makes it a reasonable pattern.

Reply:
I will for nor change it like you said.